src/Adapter/Product/QueryHandler/GetProductCombinationsForEditingHandler.php

```diff
@@ -0,0 +1,149 @@
+        $this->productRepository->assertProductExists($productId);
+        $combinations = $this->combinationRepository->getProductCombinations($productId, $query->getLimit(), $query->getOffset());
+
+        $combinationIds = array_map(function ($combination): int {
+            return (int) $combination['id_product_attribute'];
+        }, $combinations);
+
+        $attributesInformation = $this->combinationRepository->getAttributesInfoByCombinationIds(
+            $combinationIds,
+            $query->getLanguageId()
+        );
+
+        return $this->formatCombinationsForEditing(
+            $combinations,
+            $attributesInformation,
+            $this->combinationRepository->getTotalCombinationsCount($productId)
+        );
+    }
+
+    /**
+     * @param array $combinations
+     * @param array<int, array<int, mixed>> $attributesInformationByCombinationId
+     * @param int $totalCombinationsCount
+     *
+     * @return CombinationListForEditing
+     */
+    private function formatCombinationsForEditing(
+        array $combinations,
+        array $attributesInformationByCombinationId,
+        int $totalCombinationsCount
+    ): CombinationListForEditing {
+        $combinationsForEditing = [];
+
+        foreach ($combinations as $combination) {
+            $combinationId = (int) $combination['id_product_attribute'];
+            $combinationAttributesInformation = [];
+
+            foreach ($attributesInformationByCombinationId[$combinationId] as $attributesInfo) {
+                foreach ($attributesInfo as $attributeInfo) {
+                    $combinationAttributesInformation[] = new CombinationAttributeInformation(
+                        (int) $attributeInfo['id_attribute_group'],
+                        $attributeInfo['attribute_group_name'],
+                        (int) $attributeInfo['id_attribute'],
+                        $attributeInfo['attribute_name']
+                    );
+                }
+            }
+
+            $combinationsForEditing[] = new CombinationForEditing(
+                $combinationId,
+                $this->buildCombinationName($combinationAttributesInformation),
+                $combinationAttributesInformation
+            );
+        }
+
+        return new CombinationListForEditing($totalCombinationsCount, $combinationsForEditing);
+    }
+
+    /**
+     * @param CombinationAttributeInformation[] $attributesInformation
+     *
+     * @return string
+     */
+    private function buildCombinationName(array $attributesInformation): string
+    {
+        $combinedNameParts = [];
+        foreach ($attributesInformation as $combinationAttributeInformation) {
+            $combinedNameParts[] = sprintf(
+                '%s - %s',
+                $combinationAttributeInformation->getAttributeGroupName(),
+                $combinationAttributeInformation->getAttributeName()
+            );
+        }
+
+        return implode(', ', $combinedNameParts);
```

Comment:
Please change the `,` by the configuration option : PS_ATTRIBUTE_ANCHOR_SEPARATOR

Reply:
nice catch, thanks

Reply:
I checked.
So previous page didn't use this option when building name
CombinationDataProvider L188:
```
 private function getCombinationName($attributesCombinations)
    {
        $name = [];

        foreach ($attributesCombinations as $attribute) {
            $name[] = $attribute['group_name'] . ' - ' . $attribute['attribute_name'];
        }

        return implode(', ', $name);
    }
```
It seems that one might only be used in FO.
Btw it would change the `-` between attribute group and attribute ( but not the comma).

I wonder should we start using that option in BO as well?
maybe @MatShir could answer?
btw @Progi1984 , @matks would it be a BC break to if we start generating name according to that option in BO? (it doesn't look like one, but maybe someone is using the name to parse attributes or so..)

Reply:
Wow : it's weird to have a separator in front office and a different one in back office. But I let @MatShir decide.

Reply:
IMHO this is a parameter used for format display, it shouldn't be a BC break if we start using it in both BO and FO, there might be someone that parses this but it's not a public contract we ensure that `-` would ALWAYS and STRICTLY be used so that people can parse this data So not a BC break in my opinion, and definitely a godd thing to uniformise both BO and FO display

Reply:
@jolelievre In addition to this, that should fix some bugs relative to a configuration not applied ;)

Reply:
created reminder issue to avoid blocking PR for this.
https://github.com/PrestaShop/PrestaShop/issues/21788